Tasks/Common/kubernetes-common-v2/image-metadata-helper.ts

```diff
@@ -0,0 +1,189 @@
+import * as tl from "azure-pipelines-task-lib/task";
+import * as util from "util";
+
+const matchPatternForImageName = new RegExp(/\:\/\/(.+?)\@/);
+const matchPatternForDigest = new RegExp(/\@sha256\:(.+)/);
+const matchPatternForFileArgument = new RegExp(/-f\s|-filename\s/);
+const matchPatternForServerUrl = new RegExp(/Kubernetes master is running at (.+)/);
+const appliedConfigurationKey = "kubectl.kubernetes.io/last-applied-configuration";
+
+export function getDeploymentMetadata(deploymentObject: any, allPods: any, deploymentStrategy: string, clusterInfo: any): any {
+    let imageIds: string[] = [];
+    try {
+        let containers = deploymentObject.spec.template.spec.containers;
+        if (containers && containers.length > 0) {
+            containers.forEach(container => {
+                console.log("containerName:" + container.name);
+                // Filter all pods using the container names in this deployment,
+                // and get the imageIds from pod status
+                imageIds = getImageIdsForPodsInDeployment(container.name, allPods.items, deploymentObject.metadata.uid);
+            });
+        }
+    }
+    catch (e) {
+        // Don't fail the task if the image ID extraction fails
+        console.log("Image Ids extraction failed with exception: " + e);
+    }
+
+    let name: string = deploymentObject.metadata && deploymentObject.metadata.name ? deploymentObject.metadata.name : "";
+    let relatedUrls = [getPipelineUrl(deploymentObject), getServerUrl(clusterInfo)];
+    relatedUrls = relatedUrls.concat(getManifestFilePaths());
+
+    const metadataDetails = {
+        "Name": name,
+        "Description": getDescription(),
+        "RelatedUrl": relatedUrls,
+        "ResourceUri": imageIds,
+        "UserEmail": getUserEmail(),
```

Comment:
This is user-identifiable data; we should probably allow the user to opt-in?
cc: @vithati thoughts?

Reply:
The data is pushed only when the user sets the variable "ADD_PIPELINE_METADATA" as true. By default, we are not setting it.